src/workerd/jsg/limits.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <kj/common.h>
+
+namespace workerd::jsg {
+
+class Lock;
+
+class IsolateLimitEnforcer {
+public:
+  // Called when performing a cypto key derivation function (like pbkdf2) to determine if
+  // if the requested number of iterations is acceptable. If kj::none is returned, the
+  // number of iterations requested is acceptable. If a number is returned, the requested
+  // iterations is unacceptable and the return value specifies the maximum.
+  virtual kj::Maybe<size_t> checkKdfIterations(Lock& js, size_t iterations) = 0;
```

Comment:
This feels too API-specific to be in the JSG layer.

Can we instead stick to the pattern of other limits today, and use the LimitEnforcer from the IoContext?

I suppose there is no IoContext at startup. I don't think people are invoking PBKDF at startup, so I think it would be fine to just keep the old limit as a default at startup time. If we really wanted to solve it, though, I think we'd want to find a different way to let people get at the `Worker::Isolate` at startup, without leaking specific API details into the JSG layer.

Reply:
> I don't think people are invoking PBKDF at startup

We ought to test this to be certain. But the feedback about keeping it out of jsg is certainly fair.

Reply:
Well, we don't really have to test it if we just leave the limit the same as it was before when invoked at startup. And honestly, if people are invoking this at startup I'd want the limit to be low since they're killing their own cold start time.

Reply:
I disagree only because the current default limit is *way* below the recommended safe iteration count for pbkdf2. If we want to tie the limit enforcement to the `IoContext` then I'd much rather we introduce a compat flag that disables use of pbkdf2 at the startup going forward. Specifically for `workerd` running locally, however, since we're not enforcing other limits currently, we likely shouldn't be enforcing this one.

Reply:
I'd be fine with a compat flag that reduces the limit to zero at startup.

The argument: This algorithm is explicitly designed to burn CPU. Its safety is directly proportional to how much CPU it burns. Meanwhile, startup time is entirely deterministic, that is, if you're calling it at startup, you are always calling it on exactly the same input. You should instead compute it in advance and store the result in your script as a constant.

(Heh, if we wanted to be fancy we could record PDKDF calls during validation and cache the results in the worker bundle so we don't have to repeat the computation on the edge... but obviously it's not worth implementing that since probably no one needs it.)